src/debug_commands.c

```diff
@@ -21,6 +21,34 @@
 #include "util/workers.h"
 #include "cursor.h"
 
+
+DebugCTX debugCtx = {0};
+
+void validateDebugMode(DebugCTX *debugCtx)
+{
+  // Debug mode is enabled if any of its field is non-default
+  // Should be called after each debug command that changes the debugCtx
+
+  if (debugCtx->maxDocsTBscanned > 0)
+  {
+    debugCtx->debugMode = true;
+    return;
+  }
+  if (debugCtx->maxDocsTBscannedPause > 0)
+  {
+    debugCtx->debugMode = true;
+    return;
+  }
+  if (debugCtx->pauseBeforeScan)
+  {
+    debugCtx->debugMode = true;
+    return;
+  }
```

Comment:
Lots of code duplication, and setting to even more in the future.
Consider setting `debugCtx->debugMode` to the result of a complex condition, with each case on its own line and `||` between them